model/DataQuery.php

```diff
@@ -165,13 +173,18 @@ public function setQueriedColumns($queriedColumns) {
 	 * @return SQLSelect The finalised sql query
 	 */
 	public function getFinalisedQuery($queriedColumns = null) {
-		if(!$queriedColumns) $queriedColumns = $this->queriedColumns;
+		if(!$queriedColumns) {
+			$queriedColumns = $this->queriedColumns;
+		}
 		if($queriedColumns) {
 			$queriedColumns = array_merge($queriedColumns, array('Created', 'LastEdited', 'ClassName'));
 		}
 
+		$schema = DataObject::getSchema();
 		$query = clone $this->query;
-		$ancestorTables = ClassInfo::ancestry($this->dataClass, true);
+		$baseDataClass = $schema->baseDataClass($this->dataClass());
+		$baseIDColumn = $schema->quotedID($baseDataClass);
```

Comment:
Could also use `quotedColumn($this->dataClass(), 'ID')` and let it look up the base class.